src/Umbraco.Web/Security/AuthenticationManagerExtensions.cs

```diff
@@ -82,5 +84,36 @@ public static async Task<ExternalLoginInfo> GetExternalLoginInfoAsync(this IAuth
             }
             return GetExternalLoginInfo(await manager.AuthenticateAsync(authenticationType));
         }
+        
+        public static IEnumerable<AuthenticationDescription> GetExternalAuthenticationTypes(this IAuthenticationManager manager)
+        {
+            if (manager == null) throw new ArgumentNullException(nameof(manager));
+            return manager.GetAuthenticationTypes(d => d.Properties != null && d.Properties.ContainsKey("Caption"));
```

Comment:
Question: What is the different between `d.Properties["Caption"]` and [`d.Caption`](https://docs.microsoft.com/en-us/previous-versions/aspnet/mt152177%28v%3dvs.113%29)

Reply:
Looks like the `d.Caption` calls down into `d.Properties["Caption"]`. I ported across the MS implementation, so I assume that the Caption property was added after their implementation was written 🤷‍♂️ Either way, I've updated it 🙂